server/controllers/userController.js

```diff
@@ -83,6 +85,78 @@ class UserController {
     await addToBlacklist(token);
     return successStat(res, 204, 'message', 'No Content');
   }
+
+  /**
+    * @static
+    * @description Get authorised user's profile
+    * @param {Object} req - Request object
+    * @param {Object} res - Response object
+    * @returns {Object} object containing the user's profile
+    * @memberof UserController
+    */
+  static async userProfile(req, res) {
+    const { loggedInUser } = req;
+
+    return successStat(res, 200, 'profile', {
+      username: loggedInUser.username,
+      bio: loggedInUser.bio,
+      image: loggedInUser.image,
+      following: loggedInUser.following
+    });
+  }
+
+  /**
+    * @static
+    * @description Get another user's profile
+    * @param {Object} req - Request object
+    * @param {Object} res - Response object
+    * @returns {Object} object containing the user's profile
+    * @memberof UserController
+    */
+  static async getAuserProfile(req, res) {
+    const { username } = req.params;
+
+    const user = await models.User.findOne({ where: { username } });
+
+    if (!user) return errorStat(res, 404, 'No user found');
+
+    return successStat(res, 200, 'profile', {
+      username: user.username,
+      bio: user.bio,
+      image: user.image,
+      following: user.following
+    });
+  }
+
+  /**
+  * @static
+  * @description Edit user's profile
+  * @param {Object} req - Request object
+  * @param {Object} res - Response object
+  * @returns {Object} object containing the user's profile
+  * @memberof UserController
+  */
+  static async editProfile(req, res) {
+    const { id } = req.loggedInUser;
+    const { bio, email, username } = req.body;
```

Comment:
Great Job @Hadeneekeh. I actually knew you could pull this off. I'd love to have a sync with you later to understand how you implemented the image upload successfully. I have a question though, how about if as a user, I want to update just my `email` or `username`? Can this handle it?

Reply:
Thank you for asking. Yes, it will update the database with only the input that you supplied.